var/spack/repos/builtin/packages/sgpp/package.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Sgpp(SConsPackage):
+    """SGpp is a library and framework for sparse grids in different flavors.
+    SGpp supports both hierarchical spatially-adaptive sparse grids and the
+    dimensionally-adaptive sparse grid combination technique."""
+
+    homepage = "https://sgpp.sparsegrids.org"
+    url = "https://github.com/SGpp/SGpp/archive/v3.2.0.tar.gz"
+    maintainers = ['G-071', 'leiterrl', 'pfluegdk']
+
+    # Versions with Python 3 bindings:
+    version('master', git='https://github.com/SGpp/SGpp.git', branch='master')
+    version('3.3.0', sha256='ca4d5b79f315b425ce69b04940c141451a76848bf1bd7b96067217304c68e2d4')
+    version('3.2.0', sha256='dab83587fd447f92ed8546eacaac6b8cbe65b8db5e860218c0fa2e42f776962d')
+    # Versions with Python 2 bindings:
+    version('3.1.0', sha256='6b46bc5b3966e92567d6754130666bdffb7be1d1d2c1b427d7ce964b8eaab526')
+    version('3.0.0', sha256='4dd9049e664abd7db78c355fea5e192167812f443115d4bf686a51bb1e9bda9c')
+
+    # Patch that ensures libraries will actually
+    # be copied into prefix/lib upon installation
+    # (otherwise it would be prefix/sgpp/lib)
+    patch('directory.patch', when='@:3.2.0')
+    # Fix faulty setup.py in 3.2.0
+    patch('fix-setup-py.patch', when='@3.2.0')
+    # Backport opencl fix
+    patch('ocl.patch', when='@:3.2.0+opencl')
```

Comment:
Have these patches been submitted upstream?

Reply:
The patches have been submitted upstream and are part of sgpp in the versions 3.3.0 and newer (master).

However, I have talked with the current maintainer and there are currently no plans to create updated versions of the old releases. Hence, to build the sgpp spack package with the old versions, these patches will remain relevant for the spack package!